lib/beaker-pe/install/pe_utils.rb

```diff
@@ -1464,6 +1461,24 @@ def do_install_pe_with_pe_managed_external_postgres(hosts, opts)
           end
         end
 
+        #Check the lastest install log to confirm the expected failure is there
+        def is_expected_pe_postgres_failure?(host)
+          expected_failure = false
+          installer_log_dir = '/var/log/puppetlabs/installer'
+          latest_installer_log_file = on(host, "ls -1t #{installer_log_dir} | head -n1").stdout.chomp
+          # As of PE Irving (PE 2018.1.x), these are the only two expected errors
+          allowed_errors = ["The operation could not be completed because RBACs database has not been initialized",
+            "Timeout waiting for the database pool to become ready"]
+
+          allowed_errors.each do |error|
+            if(on(host, "grep '#{error}' #{installer_log_dir}/#{latest_installer_log_file}", :acceptable_exit_codes => [0,1]).exit_code == 0)
+              expected_failure = true
```

Comment:
@cthorn42 one thing that could potentially save you a lot of time executing commands on SUTs after you've already found one error would be to just `return true` here & then `return false` instead of an `expected_failure` value below.

Unless you specifically want to see combinations of expected postgres failures in install logs?

Reply:
Good catch!